### ISSUE TITLE
Consider caching package manager results

### ISSUE THREAD

**Issue by jhg03a** [petems/petems-pkg_updates#2]
Using the package manager during a puppet run can cause significant performance issues.  Instead it might be a better idea to setup a cronjob to create a cache file periodically that can just be read.  This is how the [puppet-mcollective module](https://github.com/voxpupuli/puppet-mcollective) handles caching facts.

**Comment by petems** [comment]
Indeed: https://github.com/voxpupuli/puppet-mcollective/commit/f363cd50df57339365498286a348638456c9815b 😄 

This was actually made for a specific use-case a customer asked for where Puppet was being used primarily as a reporting tool, so they didn't mind too much about run-times.

There's a more clean, official approach in https://github.com/puppetlabs/puppetlabs-package_updates, which creates a cronjob. It also can be used to make an ENC that loads all the package versions and injects them into catalogs, it's pretty neat! 👍
